matrixcomponent/JSONparser.py

```diff
@@ -35,7 +35,11 @@ def process_path(line=None):
         p = matrix.Path(path['path_name'])
 
         for b in path['bins']:
-            p.bins.append(p.Bin(b[0], b[1], b[2], b[4], b[5]))
+            ranges = b[4]
+            if type(ranges) is not list and len(b) >= 6:
+                ranges = [[b[4], b[5]]]
+
+            p.bins.append(p.Bin(b[0], b[1], b[2], ranges))
```

Comment:
@dimatr while you are at it, could you also change `p.bins` type to be `SortedDict`? This would eliminate the need to have _bin_set and finalize_bins() in Path class.

Reply:
Done, and thank you for the catch!